src/sdcard.c

```diff
@@ -303,7 +303,7 @@ uint8_t mega65_sdcard_writesector(const uint32_t sector_number)
     }
 
     // Copy the read data to a buffer for verification
-    lcopy(sd_sectorbuffer, (long)verify_buffer, 512);
+    lcopy(sd_sectorbuffer, (uint32_t)verify_buffer, 512);
```

Comment:
Nitpicking, but if you find that you need casts like this, it is often an indication that you are using inconsistent (signed/unsigned) types. In a perfect world, any needed casts in situations of low memory or I/O is best done in the lowest routine that needs to deal with the actual I/O and layers above should use consistent types.
But yeah, it may be beyond the scope of this PR.

Reply:
I fully agree that we shouldn't need to cast, but wanted to at least make the mismatch explicit in the code as e.g. cc65 doesn't warn about this. This should be fixed more long term, but I would wait until we have more tests.

Reply:
It sounds like a good idea to wait until there are more tests in place.